scripts/cybershake/auto_submit.py

```diff
@@ -608,6 +611,15 @@ def main(args, main_logger: Logger = workflow_logger.get_basic_logger()):
     else:
         args.n_runs = DEFAULT_N_RUNS
 
+    logger.debug("Processes to be run were: {}. Getting all required dependencies now.".format(args.tasks_to_run))
+    args.tasks_to_run = [const.ProcessType.get_by_name(proc) for proc in args.tasks_to_run]
+    for task in args.tasks_to_run:
+        logger.debug("Process {} in processes to be run, adding dependencies now.".format(task.str_value))
+        for proc_num in task.dependencies:
+            proc = const.ProcessType(proc_num)
+            if proc not in args.tasks_to_run:
+                args.tasks_to_run.append(proc)
```

Comment:
I presume this cascades well? I can't explicity see that here but presume that adding to the iterating list gives that cascading.

Reply:
Yep, adding to the end of the list means the for loop is dynamic and only ends when we stop adding things